smoketests/tests/add_remove_index.py

```diff
@@ -64,7 +64,7 @@ def test_add_then_remove_index(self):
         # There are no indices, resulting in an unsupported unindexed join.
         self.publish_module(name, clear = False)
         with self.assertRaises(Exception):
-            self.subscribe(self.JOIN_QUERY, n = 0)
+            self.subscribe(self.JOIN_QUERY, n = 0)()
```

Comment:
adding this `()` causes us to wait for the subscription results (which, in this case, will throw an exception).

This is the same for the lines below.